Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp

```diff
@@ -1290,9 +1293,13 @@ void OMGIRGenerator::reloadMemoryRegistersFromInstance(const MemoryInformation&
     }
 }
 
-void OMGIRGenerator::emitExceptionCheck(CCallHelpers& jit, ExceptionType type)
+void OMGIRGenerator::emitExceptionCheck(CCallHelpers& jit, Origin origin, ExceptionType type)
 {
     jit.move(CCallHelpers::TrustedImm32(static_cast<uint32_t>(type)), GPRInfo::argumentGPR1);
+    if (auto* omgOrigin = std::bit_cast<const OMGOrigin*>(origin.data())) {
+        jit.move(CCallHelpers::TrustedImm32(omgOrigin->m_callSiteIndex.bits()), GPRInfo::argumentGPR2);
+        jit.store32(GPRInfo::argumentGPR2, CCallHelpers::tagFor(CallFrameSlot::argumentCountIncludingThis));
+    }
```

Comment:
Why do you need this? Why not just have operationWasmToJSException also get the callsite from the callee? It seems like that would be a reasonable code size decrease? 

Also, can't you just do `jit.store32(CCallHelpers::TrustedImm32(omgOrigin->m_callSiteIndex.bits()), CCallHelpers::tagFor(CallFrameSlot::argumentCountIncludingThis));`? On X86_64 it might get fused.